app/core/RPCMethods/RPCMethodMiddleware.ts

```
@@ -375,6 +376,21 @@ export const getRpcMethodMiddleware = ({
         res.result = rawSig;
       },
 
+      personal_ecRecover: () => {
+        const data = req.params[0];
+        const signature = req.params[1];
+        const address = recoverPersonalSignature({ data, signature });
+
+        res.result = address;
+      },
+
+      parity_checkRequest: () => {
+        // This method is retained for legacy reasons
+        // It doesn't serve it's intended purpose anymore of checking parity requests,
+        // because our API doesn't support parity requests.
+        res.result = null;
```

Should we return an error here instead? That way we can give a useful message about this methods removal / deprecation (if anyone is even using it)

---

My intent here was just to preserve the pre-existing behavior, and leave the mobile team and/or your team to decide what to do with it from here on out.

Though, with the new permission controller update, this method may have been broken already :thinking: I don't see it listed as an unrestricted method, which would mean it would no longer work. So maybe this decision has already been made. I'll follow up on this.

---

It looks like the permission system is not yet used for WalletConnect or SDK, so I'll assume we still want to preserve these for now.